packages/server/api/controllers/products.controller.js

```diff
@@ -40,16 +40,25 @@ const getProductsByCategory = async (category) => {
         'products.price',
       )
       .leftJoin('categories', 'products.category_id', 'categories.id')
-      .where('categories.name', 'like', `${category}`);
+      .where('categories.name', 'like', `${category}`)
+      .limit(limit)
+      .offset(offset);
     if (products.length === 0) {
       throw new HttpError(
         `There are no products available with this category ${category}`,
         404,
       );
     }
-    return products;
+    return {
+      status: 200,
+      data: products,
+    };
   } catch (error) {
-    return error.message;
+    return {
+      status: error.httpStatus,
+      data: [],
+      message: error.message,
+    };
```

Comment:
This controller now mixes error handling patterns, here we return an error response, but on line 47 we throw a HttpError, wouldn't it be better to use one style?

Reply:
First it was the only http error I have changed in the return to object passing the details to become easier when I fetch the data in component

Reply:
Right, but it would be best if every endpoint acted the same way when an error happens.